xchange-stream-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingService.java

```diff
@@ -252,4 +269,50 @@ public void channelInactive(ChannelHandlerContext ctx) {
       }
     }
   }
+
+  @Override
+  public void resubscribeChannels() {
+    if (isPrivate) {
+      super.resubscribeChannels();
+    } else {
+      subscriptionRequestMap.clear();
+      channelIds.clear();
+      HashMap<String, KrakenSubscriptionMessage> messages = new HashMap<>();
+      for (Map.Entry<String, Subscription> entry : super.channels.entrySet()) {
+
+        String[] channelData = entry.getKey().split(KRAKEN_CHANNEL_DELIMITER);
```

Comment:
nit: this would be easier to read if maybe you created a `KrakenSubscription` object and added a decode method, e.g.

```
private KrakenSubscription decodeSubscription(Subscription subscription) {}
```